### ISSUE TITLE
[@typescript-eslint/consistent-indexed-object-style] Fails on circular references

### ISSUE THREAD

**Issue by papb** [typescript-eslint/typescript-eslint#2687]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-indexed-object-style": ["error"]
  }
}
```

```TS
type Tree = {
	[key: string]: number | Tree;
};
```

**Expected Result**

Since it is not possible to convert that to a Record, the rule should not trigger.

```ts
type Tree = Record<string, number | Tree>; // Type alias 'Tree' circularly references itself.
```

**Actual Result**

The rule triggered.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | latest |
| `@typescript-eslint/parser`        | latest |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | latest |
| `node`                             | 12 |

**Comment by lakshyabatman** [comment]
@papb I'm interested to work on this issue, so I have to put up a condition to check if the value type is same as the node type. 
Right?

**Comment by papb** [comment]
Hi @lakshyabatman! Thank you! I think so, yes.

**Comment by Paril** [comment]
Just noting that this issue is still alive and well :)

**Comment by bradzacher** [comment]
With any issue in opened in this project - it either has a visible progress in the form of an attached PR, or it has no progress.

We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.

If not - please just subscribe to the issue and wait patiently.
Your comment does not bump issue priority in any way and just serves to spam ***everyone*** subscribed to the issue.

**Comment by jp7837** [comment]
I've linked the issue above in the TypeScript repo which is currently still open.
